jax/_src/numpy/lax_numpy.py

```diff
@@ -893,15 +893,17 @@ def isrealobj(x: Any) -> bool:
   return not iscomplexobj(x)
 
 
-def reshape(a: ArrayLike, newshape: DimSize | Shape, order: str = "C") -> Array:
+def reshape(
+    a: ArrayLike, shape: DimSize | Shape, order: str = "C", *,
```

Comment:
I think we (temporarily) need a default for `shape`. The reason is that if someone does `jnp.reshape(x, newshape=shape)` they won't see the deprecation warning, they'll see `TypeError: reshape() missing 1 required positional argument: 'shape'`